### ISSUE TITLE
Error when sampling using HMASynthesizer

### ISSUE THREAD

**Issue by JanJacekJaniszewski** [sdv-dev/SDV#1603]
### Environment Details
Please indicate the following details about the environment in which you found the bug:

* SDV version: 1.4.0
* Python version: 3.9
* Operating System: MAC Venture 13.5.2 (22G91)

### Error Description
When I set the distributions in the datasets, I get an error indicating something with TruncatedGaussian not working as distribution. However, I did not even define this distribution anywhere.

#### Input
```
from sdv.multi_table import HMASynthesizer
from sdv.datasets.demo import download_demo

real_data, real_metadata = download_demo(
    modality='multi_table',
    dataset_name='CORA_v1'
)

synthesizer = HMASynthesizer(real_metadata)

# CONTENT TABLE
synthesizer.set_table_parameters(
    table_name='content',
    table_parameters={
        'enforce_min_max_values': True,
        'default_distribution': 'truncnorm',
        'numerical_distributions': {
            'add_numerical': 'truncnorm' }
    }
)

# PAPER TABLE
synthesizer.set_table_parameters(
    table_name='paper',
    table_parameters={
        'enforce_min_max_values': True,
        'default_distribution': 'gamma',
        'numerical_distributions': {
            'sum(cited_paper_id)': 'gamma',
            'max(cited_paper_id)': 'gamma',
            'max(citing_paper_id)': 'gamma',
            'add_numerical': 'gamma',
            'nb_rows_in_cites': 'gamma',
            'nb_rows_in_content': 'gamma',
            'sum(citing_paper_id)': 'gamma',
            'min(citing_paper_id)': 'gamma',
            'min(cited_paper_id)': 'gamma' }
    }
)

synthesizer.fit(real_data)

synthetic_data = synthesizer.sample(scale=1)
```

#### Output
```
Preprocess Tables: 100%|██████████| 3/3 [00:00<00:00, 13.44it/s]

Learning relationships:
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):   3%|▎         | 44/1565 [00:00<00:28, 52.68it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):   8%|▊         | 123/1565 [00:02<00:28, 50.84it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):  15%|█▍        | 228/1565 [00:04<00:23, 56.19it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):  16%|█▋        | 255/1565 [00:04<00:22, 58.66it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):  24%|██▍       | 378/1565 [00:07<00:19, 59.56it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):  41%|████      | 641/1565 [00:11<00:14, 62.47it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):  42%|████▏     | 665/1565 [00:11<00:13, 64.80it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):  65%|██████▍   | 1017/1565 [00:16<00:06, 81.00it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'):  67%|██████▋   | 1045/1565 [00:16<00:06, 81.27it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(1/3) Tables 'paper' and 'cites' ('cited_paper_id'): 100%|██████████| 1565/1565 [00:21<00:00, 71.61it/s] 
(2/3) Tables 'paper' and 'cites' ('citing_paper_id'):   9%|▉         | 195/2222 [00:03<00:42, 47.83it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(2/3) Tables 'paper' and 'cites' ('citing_paper_id'):  46%|████▌     | 1014/2222 [00:19<00:20, 59.29it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/multivariate/gaussian.py:119: UserWarning: Unable to fit to a <class 'copulas.univariate.beta.BetaUnivariate'> distribution for column add_numerical. Using a Gaussian distribution instead.
  warnings.warn(warning_message)
(2/3) Tables 'paper' and 'cites' ('citing_paper_id'): 100%|██████████| 2222/2222 [00:35<00:00, 63.48it/s] 
(3/3) Tables 'paper' and 'content' ('paper_id'):   2%|▏         | 48/2708 [00:00<00:11, 236.22it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):   3%|▎         | 72/2708 [00:00<00:11, 231.83it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):   5%|▌         | 145/2708 [00:00<00:10, 238.46it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  19%|█▊        | 505/2708 [00:02<00:09, 233.28it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: invalid value encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  20%|██        | 554/2708 [00:02<00:09, 232.64it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  30%|███       | 813/2708 [00:03<00:08, 219.93it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  33%|███▎      | 885/2708 [00:03<00:07, 229.97it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  45%|████▌     | 1222/2708 [00:05<00:06, 229.80it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: invalid value encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  46%|████▌     | 1246/2708 [00:05<00:06, 232.15it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  49%|████▊     | 1319/2708 [00:05<00:06, 231.36it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: invalid value encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  50%|████▉     | 1343/2708 [00:05<00:05, 231.45it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: invalid value encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  51%|█████     | 1368/2708 [00:06<00:05, 235.06it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  58%|█████▊    | 1564/2708 [00:06<00:05, 196.95it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: invalid value encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  60%|██████    | 1638/2708 [00:07<00:04, 224.78it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: invalid value encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  63%|██████▎   | 1711/2708 [00:07<00:04, 229.90it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  64%|██████▍   | 1737/2708 [00:07<00:04, 236.11it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  69%|██████▊   | 1860/2708 [00:08<00:03, 234.82it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  70%|███████   | 1909/2708 [00:08<00:03, 237.25it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: invalid value encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  75%|███████▍  | 2029/2708 [00:08<00:02, 230.52it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  79%|███████▊  | 2126/2708 [00:09<00:02, 239.33it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  83%|████████▎ | 2249/2708 [00:09<00:02, 218.04it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  92%|█████████▏| 2497/2708 [00:10<00:00, 245.68it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  95%|█████████▌| 2573/2708 [00:11<00:00, 241.96it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  97%|█████████▋| 2624/2708 [00:11<00:00, 245.65it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: invalid value encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: divide by zero encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'):  99%|█████████▊| 2674/2708 [00:11<00:00, 240.14it/s]/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:45: RuntimeWarning: divide by zero encountered in scalar divide
  a = (self.min - loc) / scale
/Users/Jan_Janiszewski/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/copulas/univariate/truncated_gaussian.py:46: RuntimeWarning: invalid value encountered in scalar divide
  b = (self.max - loc) / scale
(3/3) Tables 'paper' and 'content' ('paper_id'): 100%|██████████| 2708/2708 [00:11<00:00, 229.26it/s]

Modeling Tables: 100%|██████████| 1/1 [00:00<00:00,  1.18it/s]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[3], line 57
     54 synthesizer.fit(real_data) # Add your data here (perhaps you have to load it first)
     56 # Generate a sample now (scale is the multiplier for how many samples should be generated; i.e., len(synthetic_data) = scale * len(YOURDATA)
---> 57 synthetic_data = synthesizer.sample(scale=1)

File ~/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/sdv/multi_table/base.py:357, in BaseMultiTableSynthesizer.sample(self, scale)
    353     raise SynthesizerInputError(
    354         f"Invalid parameter for 'scale' ({scale}). Please provide a number that is >0.0.")
    356 with self._set_temp_numpy_seed():
--> 357     sampled_data = self._sample(scale=scale)
    359 return sampled_data

File ~/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/sdv/sampling/hierarchical_sampler.py:193, in BaseHierarchicalSampler._sample(self, scale)
    191         num_rows = int(self._table_sizes[table] * scale)
    192         synthesizer = self._table_synthesizers[table]
--> 193         self._sample_table(
    194             synthesizer=synthesizer,
    195             table_name=table,
    196             num_rows=num_rows,
    197             sampled_data=sampled_data
    198         )
    200 added_relationships = set()
    201 for relationship in self.metadata.relationships:

File ~/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/sdv/sampling/hierarchical_sampler.py:136, in BaseHierarchicalSampler._sample_table(self, synthesizer, table_name, num_rows, sampled_data)
    134 if child_name not in sampled_data:
    135     for _, row in table_rows.iterrows():
--> 136         self._add_child_rows(
    137             child_name=child_name,
    138             parent_name=table_name,
    139             parent_row=row,
    140             sampled_data=sampled_data
    141         )

File ~/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/sdv/sampling/hierarchical_sampler.py:101, in BaseHierarchicalSampler._add_child_rows(self, child_name, parent_name, parent_row, sampled_data)
     99 foreign_key = self.metadata._get_foreign_keys(parent_name, child_name)[0]
    100 num_rows = self._get_num_rows_from_parent(parent_row, child_name, foreign_key)
--> 101 child_synthesizer = self._recreate_child_synthesizer(child_name, parent_name, parent_row)
    103 sampled_rows = self._sample_rows(child_synthesizer, num_rows)
    105 if len(sampled_rows):

File ~/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/sdv/multi_table/hma.py:274, in HMASynthesizer._recreate_child_synthesizer(self, child_name, parent_name, parent_row)
    271 table_meta = self.metadata.tables[child_name]
    273 synthesizer = self._synthesizer(table_meta, **self._table_parameters[child_name])
--> 274 synthesizer._set_parameters(parameters)
    275 synthesizer._data_processor = self._table_synthesizers[child_name]._data_processor
    277 return synthesizer

File ~/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/sdv/single_table/copulas.py:378, in GaussianCopulaSynthesizer._set_parameters(self, parameters)
    375     self._num_rows = 0 if pd.isna(num_rows) else max(0, int(round(num_rows)))
    377 if parameters:
--> 378     parameters = self._rebuild_gaussian_copula(parameters)
    379     self._model = multivariate.GaussianMultivariate.from_dict(parameters)

File ~/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/sdv/single_table/copulas.py:343, in GaussianCopulaSynthesizer._rebuild_gaussian_copula(self, model_parameters)
    341 for column, univariate in model_parameters['univariates'].items():
    342     columns.append(column)
--> 343     univariate['type'] = self.get_distribution_class(
    344         self._numerical_distributions.get(column, self.default_distribution)
    345     )
    346     if 'scale' in univariate:
    347         univariate['scale'] = max(0, univariate['scale'])

File ~/McK_Internal/ai-powered-dataops/venv/lib/python3.9/site-packages/sdv/single_table/copulas.py:89, in GaussianCopulaSynthesizer.get_distribution_class(cls, distribution)
     87 if not isinstance(distribution, str) or distribution not in cls._DISTRIBUTIONS:
     88     error_message = f"Invalid distribution specification '{distribution}'."
---> 89     raise ValueError(error_message)
     91 return cls._DISTRIBUTIONS[distribution]

ValueError: Invalid distribution specification '<class 'copulas.univariate.truncated_gaussian.TruncatedGaussian'>'.
```

**Comment by npatki** [comment]
Hi @JanJacekJaniszewski, thank for filing with all the details.

Right now, the HMASynthesizer does not support schemas that have >1 connections between the same tables. So if you're exploring the SDV `CORA_v1` may not be the best place to start. (Note that our other synthesizers such as [HSA](https://docs.sdv.dev/sdv/multi-table-data/modeling/synthesizers/hsasynthesizer) do support this schema.) There is an outstanding request for this in #1508, if you'd like to follow along.

As for the `ValueError`: The `TruncatedGaussian` is the distribution that `'truncnorm'` is supposed to be referring to. We have investigated this a bit and found that the error is specific to:

1. HMASynthesizer (other such as HSA and Independent are working fine),
2. The `numerical_distributions` parameter (the `default_distribution` is working fine), AND
3. Child tables, which are tables that have foreign keys (other tables are working fine)

I have filed a fresh issue for this in #1605, where I've replicated in on a supported schema (`fake_hotels`). I suggest we can move our conversation there.
